### PR TITLE
Auto infer device for SPLADE

### PR DESCRIPTION
**Pull request by acatav**
## Problem

Device currently not inferred automatically for SPLADE 

## Solution

Use torch to automatically infer device, and allow override

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Infrastructure change (CI configs, etc)
- [X] Non-code change (docs, etc)
- [ ] None of the above: (explain here)

## Test Plan

Added unit tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -104,6 +104,11 @@ from pinecone_text.sparse import SpladeEncoder
 # Initialize Splade
 splade = SpladeEncoder()
 
+# Or with "cuda" in case it's avialable
+# import pytorch
+# device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
```
Comment (igiloh-pinecone) [inline README.md:109]: This code should simply happen inside `SpladeEncoder.__init__()`, the user doesn't need to explicitly pass the device.

**Review by igiloh-pinecone — CHANGES_REQUESTED** [review]
@acatav I don't see why the user would need to manually pass in a device. This can simoly happen automatically

File: README.md (hunk 1 above)
Comment (acatav) [reply README.md:109]: but what if he have cuda, but don't want to use it? for example, maybe for queries you don't want to

Comment (acatav) [reply README.md:109]: Anyway, I don't want to break API right now

Comment (acatav) [reply README.md:109]: I'll resolve this to not let this PR hang too much

Comment (acatav) [reply README.md]: done

**Review by igiloh-pinecone — APPROVED** [review]
LGTM

**Pull request merged**